server/datasources/requestsDataSource.ts

```diff
@@ -1,21 +1,24 @@
 import { DataSource } from 'apollo-datasource'
+import dotenv from 'dotenv'
 import { Types } from 'mongoose'
 
 import { Request, RequestDocument, RequestInterface } from '../models/requestModel'
-import { config } from '../config'
 import { RequestsCache } from '../database/cache'
 
+dotenv.config()
+const CACHING = process.env.CACHING == 'TRUE'
+
 export default class RequestDataSource extends DataSource {
   async getRequestById(rawId: string): Promise<RequestInterface> {
     const id = Types.ObjectId(rawId)
     let result
 
-    if (config.caching) {
+    if (CACHING) {
       result = RequestsCache.getData().filter(request => request._id.equals(id))[0]
     } else {
-      Request.findById(id).exec()
+      await Request.findById(id).exec()
```

Comment:
Do we still need the await keyword if we're chaining a .then() after?

Reply:
I added `await` because otherwise there's no guarantee that `result = res` will execute before `return this.requestReducer(result)` on line 28.

Reply:
Makes sense